### PR TITLE
Add Azure subscriptions and management groups

### PR DESCRIPTION
**Pull request by teszes**
Progress towards #50

### CODE REVIEW

**Comment by teszes** [comment]
Checks not passing because of the changeme naming requirement. It is not exactly clear to me what is the point of those, doesn't being a module variable adequately signal that it should be provided as a variable?

**Comment by teszes** [comment]
The exact data blocks the check has a problem with BTW probably shouldn't be changed directly, only by changing the variables themselves.

**Comment by ttarczynski** [comment]
Thank you  @teszes for your contribution!

Just to expand a bit on the naming requirements.
See the conventions documented here: [README.md > Conventions](https://github.com/ContainerSolutions/terraform-examples/blob/main/README.md#conventions)

There are two main naming requirements:
- all names that can be change by the user (resources, variables, data items) should be prefixed with `changeme_`
  - it is to indicate to users new to terraform what bits are just names and can be changed
- all names should be unique across all examples (resources, variables, data items)
  - it is to enable easy mixing of code from multiple examples together

**Comment by ttarczynski** [comment]
There's one more convention (not yet added to README):
- There should be comments explaining and linking to documentation
- Please compare:
  -[google/google_compute_instance/simple/main.tf](https://github.com/ContainerSolutions/terraform-examples/blob/main/google/google_compute_instance/simple/main.tf)

The comments that we want to have in all examples:
- `# Summary:` – 1-line summary of what this example does
- `# Documentation:` – add link to documentation before each block (terraform, provider, variable, resource, etc.)
- `# Explanation:` – add where some extra explanation is needed

**Comment by teszes** [comment]
Name collision, will update

**Comment by ttarczynski** [comment]
Thank you @teszes for the fixes.
I've merged it to the `integration` branch to run CI tests applying the azure examples:
- Test definition: [.github/workflows/main.yml#L128-L150](https://github.com/ContainerSolutions/terraform-examples/blob/main/.github/workflows/main.yml#L128-L150)
- Last test run on `integration` branch: https://github.com/ContainerSolutions/terraform-examples/runs/3022667473?check_suite_focus=true

Our `azurerm` tests stopped working after previous change #56. I've described the issue in more detail in: #69

We need to fix #69 first, to be able to run automated CI tests before merging all the Azure pull requests.

**Comment by teszes** [comment]
That's cool, and see you at the meeting, I hope I can help fixing the CI

File: azurerm/azurerm_management_group/child/main.tf
Comment (ttarczynski) [inline azurerm/azurerm_management_group/child/main.tf:22]: When using a variable without a default value it will require the user to input this value when running `./run.sh`.
It also breaks the CI "Run Azure Examples", see: https://github.com/ContainerSolutions/terraform-examples/runs/3073983061
```
Run /home/runner/work/terraform-examples/terraform-examples/bin/run_azurerm_examples.sh
azurerm/azurerm_management_group/child
./run.sh

var.changeme_parent_management_group_id
╷
│ Error: No value for required variable
│ 
│   on main.tf line 20:
│   20: variable "changeme_parent_management_group_id" ***
│ 
│ The root module input variable "changeme_parent_management_group_id" is not
│ set, and has no default value. Use a -var or -var-file command line
│ argument to provide a value for this variable.
```

We need to make this example work without user input.

----

I'm not sure about the intent here and how exactly it works in Azure. But maybe creating both `child_management_group` and `child_management_group` here (in the same `main.tf`) would help?

File: azurerm/azurerm_management_group/child/main.tf
Comment (ttarczynski) [inline azurerm/azurerm_management_group/child/main.tf:17]: This is good. Just to clarify:

Here it works without a default value, because we've taken care of it in: https://github.com/ContainerSolutions/terraform-examples/blob/integration/bin/shared_azurerm.sh
So this variable is the only one for the `azurerm` provider to be passed via environment variables (eg. in CI tests).

**Review by ttarczynski — COMMENTED** [review]
One fix needed (see inline comments). All the rest looks good to me.
One more hint on testing locally. You can run the CI apply test locally with:
```
export ARM_SUBSCRIPTION_ID=<YOUR_AZURE_SUBSCRIPTION_ID>
./bin/run_azurerm_examples.sh
```

File: azurerm/azurerm_management_group/child/main.tf
Comment (teszes) [reply azurerm/azurerm_management_group/child/main.tf:22]: TBH it would need to be provided as the MG depends on this value and there is no universal default solution for this.

We have to do sg with this if it breaks CI though. What if I disabled CI for this use case, and we find out how to handle examples when they have dependencies?

File: azurerm/azurerm_management_group/child/main.tf
Comment (teszes) [reply azurerm/azurerm_management_group/child/main.tf:17]: I see, I'd like to create a similar solution for the other providers, I found out how we could test the other Azure providers, we don't have to leave anything untested :)

Comment (ttarczynski) [reply azurerm/azurerm_management_group/child/main.tf:17]: LGTM!

File: azurerm/azurerm_management_group/child/main.tf
Comment (ttarczynski) [reply azurerm/azurerm_management_group/child/main.tf:22]: Could we do something like:

```
resource "azurerm_management_group" "changeme_parent_management_group" {
  display_name = "changeme-parent-management-group-display-name"
}

resource "azurerm_management_group" "changeme_child_management_group" {
  display_name               = "changeme-child-management-group-display-name"
  parent_management_group_id = azurerm_management_group.changeme_parent_management_group.id
}
```

So, to create both resources here, instead of having dependencies between the examples. 
What do you think?

Comment (teszes) [reply azurerm/azurerm_management_group/child/main.tf:22]: will do definitely, I want to rebase each branch as they are merged, so this will wait until we merge the rest

**Review by ttarczynski — COMMENTED** [review]
I've added inline comments where fixes or some clarification are needed.

**Comment by ttarczynski** [comment]
Added one small fix: https://github.com/ContainerSolutions/terraform-examples/pull/57/commits/719384efae48b556315bb7c392eae5400dc1f903 , run CI apply test on integration, and ready to merge 👍

**Review by ttarczynski — APPROVED** [review]
LGTM

**Pull request merged**
